tests/src/data/use-cases/pet/db-appoint-specie.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { type LoadSpecieByNameRepository } from '@/data/protocols'
+import { DbAppointSpecie } from '@/data/use-cases'
+import { type AppointSpecie } from '@/domain/use-cases'
+import { makeFakeSpecieRepository } from '@/tests/utils'
+
+interface SutTypes {
+  sut: DbAppointSpecie
+  specieRepositoryStub: LoadSpecieByNameRepository
+}
+
+const makeSut = (): SutTypes => {
+  const specieRepositoryStub = makeFakeSpecieRepository()
+  const dependencies: AppointSpecie.Dependencies = {
+    specieRepository: specieRepositoryStub
+  }
+  const sut = new DbAppointSpecie(dependencies)
+
+  return {
+    sut,
+    specieRepositoryStub
+  }
+}
+
+describe('DbAppointOtherSpecie Use Case', () => {
+  const specieName: AppointSpecie.Params = 'any_name'
+
+  describe('LoadSpecieByName', () => {
+    it('Should return specie equal other and specieAlias equal specieName when specieName is not equal to specie in db', async () => {
+      const { sut, specieRepositoryStub } = makeSut()
+      const otherSpecie = {
+        id: 'any_id',
+        name: 'Outros'
+      }
+      jest.spyOn(specieRepositoryStub, 'loadByName')
+        .mockResolvedValueOnce(undefined)
+        .mockResolvedValueOnce(otherSpecie)
```

Comment:
why are you mocking up two returns?

Reply:
because real scenario flow this returns make sense. Note that "Outros" specieName will cause an weird case. Other specie with null alias, my solution is not too elegant, but make sense

Reply:
make sense to me.